### DISCUSSION TITLE
Not getting app extracted for installation

### DISCUSSION THREAD

**Discussion by jach77** [grahampugh/erase-install#263] [Q&A] [answered]
I can't seem to get the .app extracted for the actual install. I am using Jamf Pro as documented here. I have tried a combination of parameters, all with the same result. 

Initial Parameters
P4: --force-curl 
P5: --update
P6: --os=12.4
P7 --move

Current parameters
P4: --force-curl 
P5: --pkg
P6: --os=12.4
P7 --move

It seems everything works up to the --move parameter. I've run through the process about half a dozen times. Here is the output:

17  012-06873       12.4       21F79    2022-05-24  macOS Monterey                 
18  012-22320       11.6.7     20G630   2022-06-16  macOS Big Sur                  

Build 21F79 selected. Downloading #17...

Making empty sparseimage...
*********************************************************
*** Working around a very dumb Apple bug in a package ***
*** postinstall script that fails to correctly target ***
*** the Install macOS.app when installed to a volume  ***
*** other than the current boot volume.               ***
***       Please file feedback with Apple!            ***
*********************************************************
Product downloaded and installed to /Library/Management/erase-install/Install_macOS_12.4-21F79.sparseimage
   [run_installinstallmacos] Mounting sparse disk image to identify installer app.
   [erase-install] Installer is at: /Volumes/Install_macOS_12.4-21F79/Applications/Install macOS Monterey.app
   [erase-install] Closing jamfHelper download message (language=en)

   [erase-install] Mounted installer will be unmounted: /Volumes/Install_macOS_12.4-21F79/Applications/Install macOS Monterey.app
Volume Install_macOS_12.4-21F79 on disk3s2 force-unmounted
   [erase-install] Cleaning working directory '/Library/Management/erase-install/content'

   [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success
/Library/Management/erase-install/erase-install.sh: line 1079:   975 Terminated: 15          /usr/bin/caffeinate -dimsu -w $$

I appreciate any help you can provide!!

**Comment by grahampugh** [comment, marked as answer]
For some reason it looks like it is not seeing your entry --move. Not sure why from your output. Got any spaces in there? 

By the way --os=12.4 is not a valid parameter. (I'm not sure why this seems to confuse so many people). The --os parameter is for major releases, e.g. 10.15, 11, 12.  The --version parameter is for specific minor versions, e.g. 10.15.7, 12.4, 11.6.6. 

At the moment you are lucky that you are getting 12.4 anyway, because that's the latest release, which is what you get without any --os, --version or --build parameters. 

As a last comment, it's easier to just use the package and use the run command to supply parameters. See the wiki section for Jamf Pro.

**Reply by jach77** [reply]
Thank you thank you thank you. Your documentation is very clear, and I appreciate that. What ended up working for me is the Jamf option #1. For whatever reason, I skipped right over that to #2. This is what I am using. 

/Library/Management/erase-install/erase-install.sh --force-curl --version=12.4 --reinstall --update --move --depnotify --check-power

**Reply by grahampugh** [reply]
Glad it's working. A couple of points. You don't need --force-curl with the package because the correct version of installinstallmacos is included.  Also --update and --version are incompatible options - --update is designed to keep you to the latest available installer but you have selected a specific one.
